File: web/src/layout/MapDemoXml.ts
//
// MapDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import htmlForDefault from './map_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.MapDemoXml
export class MapDemoXml {
    xmlRoot!: HTMLElement;
    map!: HTMLDivElement;
    select!: HTMLInputElement;
    options!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.map = getViewById<HTMLDivElement>(view, "map");
        this.select = getViewById<HTMLInputElement>(view, "select");
        this.options = getViewById<HTMLDivElement>(view, "options");
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/MapDemoXml.kt
//
// MapDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class MapDemoXml {

    lateinit var map: com.google.android.gms.maps.MapView
    lateinit var select: EditText
    lateinit var options: androidx.recyclerview.widget.RecyclerView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.map_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        map = view.findViewById<com.google.android.gms.maps.MapView>(R.id.map)
        select = view.findViewById<EditText>(R.id.select)
        options = view.findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.options)
        
        
        
        return view
    }
}